Source/WebKit/UIProcess/API/haiku/WebView.cpp

```diff
@@ -63,6 +63,8 @@ BWebView::BWebView(BRect frame, BWindow* myWindow)
 
     fViewPort=adoptWK(WKViewCreate("Webkit", frame, myWindow, config.get()));
 
+    // TODO: Can we run WebKit's main thread on its own thread instead of on
```

Comment:
why would we want to do that? Isn't that an extra thread for no reason?

Reply:
The main thread will be a BApplication, if WebKit can attach itself (as a BHandler) to that application, it is probably a better idea to do that in the end.

Other cases where we had threads running "in the background" without really telling the application about them did not go very well, it can quickly become very complex to coordinate the shutdown of these threads, for example.

Also, more generally, I don't think embedding a web engine in an application just for an help system is a great idea. The app can distribute the help files as html and open a web browser to render them. There are other use cases for embedding HTML (such as email readers), but we will see how to handle that (it may be a good idea in that case to completely disable network access and javascript, for example, or at least strongly sandbox them).